File: java/PalindromeNumber.java
public class PalindromeNumber {
    public boolean isPalindrome(int x) {
        /**
         * Forget the negative numbers.
         */
        if (x < 0) {
            return false;
        }
        int t = x;
        long v = 0;
        while (t != 0) {
            v = v * 10 + t % 10;
            t /= 10;
        }
        return v == (long) x;
    }
}

File: cc/palindrome_number.cc
#include "base_include.h"

class PalindromeNumber {
public:
    bool isPalindrome(const int &x) {
        if (x < 0) {
            return false;
        }
        int res = 0;
        int t = x;
        while (t != 0) {
            res = res * 10 + t % 10;
            t /= 10;
        }
        return (res == x);
    }
};

int main(int argc, char *argv[]) {
    PalindromeNumber solver;
    cout << solver.isPalindrome(1) << endl;
    return 0;
}
